BLD: Flaky CI

# PyPy Failures
PyPy 3.7 randomly seems to timeout, this is often fixed by force pushing (but that's not great)

Sample error:

```bash
___________________ ERROR at setup of test_run_build_failure ___________________

tmpdir = local('/tmp/pytest-of-runner/pytest-0/test_run_build_failure0')
dummy_packages = None

    @pytest.fixture
    def basic_conf(tmpdir, dummy_packages):
>       return generate_basic_conf(tmpdir)

/home/runner/work/asv/asv/test/conftest.py:196: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/home/runner/work/asv/asv/test/conftest.py:75: in generate_basic_conf
    subdir=repo_subdir).path
/home/runner/work/asv/asv/test/tools.py:382: in generate_test_repo
    copy_template(template_path, project_path, dvcs, mapping)
/home/runner/work/asv/asv/test/tools.py:327: in copy_template
    dvcs.add(dst_path)
/home/runner/work/asv/asv/test/tools.py:192: in add
    self.run_git(['add', relpath(filename, self.path)])
/home/runner/work/asv/asv/test/tools.py:171: in run_git
    [self._git] + args, **kwargs)
/home/runner/work/asv/asv/asv/util.py:535: in check_output
    proc = subprocess.Popen(args, **kwargs)
/opt/hostedtoolcache/PyPy/3.7.13/x64/lib-python/3/subprocess.py:815: in __init__
    restore_signals, start_new_session)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <subprocess.Popen object at 0x0000000007547e50>
args = ['/usr/bin/git', 'add', 'setup.py'], executable = b'/usr/bin/git'
preexec_fn = <function check_output.<locals>.<lambda> at 0x0000000007642a20>
close_fds = True, pass_fds = ()
cwd = '/tmp/pytest-of-runner/pytest-0/test_run_build_failure0/test_repojyurescx'
env = None, startupinfo = None, creationflags = 0, shell = False, p2cread = -1
p2cwrite = -1, c2pread = 524, c2pwrite = 590, errread = 592, errwrite = 593
restore_signals = True, start_new_session = False

    def _execute_child(self, args, executable, preexec_fn, close_fds,
                       pass_fds, cwd, env,
                       startupinfo, creationflags, shell,
                       p2cread, p2cwrite,
                       c2pread, c2pwrite,
                       errread, errwrite,
                       restore_signals, start_new_session):
        """Execute program (POSIX version)"""
    
        if isinstance(args, (str, bytes)):
            args = [args]
        else:
            args = list(args)
    
        if shell:
            # On Android the default shell is at '/system/bin/sh'.
            unix_shell = ('/system/bin/sh' if
                      hasattr(sys, 'getandroidapilevel') else '/bin/sh')
            args = [unix_shell, "-c"] + args
            if executable:
                args[0] = executable
    
        if executable is None:
            executable = args[0]
        orig_executable = executable
    
        # For transferring possible exec failure from child to parent.
        # Data format: "exception name:hex errno:description"
        # Pickle is not used; it is complex and involves memory allocation.
        errpipe_read, errpipe_write = os.pipe()
        # errpipe_write must not be in the standard io 0, 1, or 2 fd range.
        low_fds_to_close = []
        while errpipe_write < 3:
            low_fds_to_close.append(errpipe_write)
            errpipe_write = os.dup(errpipe_write)
        for low_fd in low_fds_to_close:
            os.close(low_fd)
        try:
            try:
                # We must avoid complex work that could involve
                # malloc or free in the child process to avoid
                # potential deadlocks, thus we do all this here.
                # and pass it to fork_exec()
    
                if env is not None:
                    env_list = []
                    for k, v in env.items():
                        k = os.fsencode(k)
                        if b'=' in k:
                            raise ValueError("illegal environment variable name")
                        env_list.append(k + b'=' + os.fsencode(v))
                else:
                    env_list = None  # Use execv instead of execve.
                executable = os.fsencode(executable)
                if os.path.dirname(executable):
                    executable_list = (executable,)
                else:
                    # This matches the behavior of os._execvpe().
                    executable_list = tuple(
                        os.path.join(os.fsencode(dir), executable)
                        for dir in os.get_exec_path(env))
                fds_to_keep = set(pass_fds)
                fds_to_keep.add(errpipe_write)
                self.pid = _posixsubprocess.fork_exec(
                        args, executable_list,
                        close_fds, tuple(sorted(map(int, fds_to_keep))),
                        cwd, env_list,
                        p2cread, p2cwrite, c2pread, c2pwrite,
                        errread, errwrite,
                        errpipe_read, errpipe_write,
                        restore_signals, start_new_session, preexec_fn)
                self._child_created = True
            finally:
                # be sure the FD is closed no matter what
                os.close(errpipe_write)
    
            # self._devnull is not always defined.
            devnull_fd = getattr(self, '_devnull', None)
            if p2cread != -1 and p2cwrite != -1 and p2cread != devnull_fd:
                os.close(p2cread)
            if c2pwrite != -1 and c2pread != -1 and c2pwrite != devnull_fd:
                os.close(c2pwrite)
            if errwrite != -1 and errread != -1 and errwrite != devnull_fd:
                os.close(errwrite)
            if devnull_fd is not None:
                os.close(devnull_fd)
            # Prevent a double close of these fds from __init__ on error.
            self._closed_child_pipe_fds = True
    
            # Wait for exec to fail or succeed; possibly raising an
            # exception (limited in size)
            errpipe_data = bytearray()
            while True:
>               part = os.read(errpipe_read, 50000)
E               Failed: Timeout >[300](https://github.com/airspeed-velocity/asv/actions/runs/4588739966/jobs/8103171230?pr=1263#step:7:301).0s
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -10,7 +10,7 @@ jobs:
       fail-fast: false
       matrix:
         os: ["ubuntu-latest"]
-        python-version: ["3.7", "3.10", "pypy-3.7", "pypy-3.8"]
+        python-version: ["3.7", "3.10", "pypy-3.8", "pypy-3.9"]
     timeout-minutes: 30
     steps:
       - uses: actions/checkout@v3

